pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -393,6 +407,7 @@ func (d *DeploymentController) render(templateName string, mi TemplateInput) ([]
 		MeshConfig:  cfg.MeshConfig,
 		Values:      cfg.Values.Map(),
 	}
+	setInputValue(input, "global.multiCluster.clusterName", features.ClusterName)
```

Comment:
we might need network as well?

Reply:
My understanding was this should also be the cluster the Gateway actually lives in though, not features.ClusterName?

For injection, we get this from the `:ENV:` stuff which is per cluster. We don't currently pass in the clsuter here today but I think we could...?

Reply:
Or I guess the broken case is just external Istiod. We don't setup this controlelr for multiple clusters, just one. So this should be good.

serverArgs.RegistryOptions.KubeOptions.ClusterID is the real source of truth though, which defaults to features.ClusterName. We could pass the cluster ID from configcontroller.go to get this?

Reply:
> we might need network as well?

I wasn't sure if we will need it or not, so I didn't add anything for network. It's not needed for the broken external istiod problem anyway. I noticed that you can set network with a Gateway label, and wasn't sure how that should be used.

> We could pass the cluster ID from configcontroller.go to get this?

Done.